### DISCUSSION TITLE
logisim in the browser?

### DISCUSSION THREAD

**Discussion by regehr** [logisim-evolution/logisim-evolution#1933] [General]
hi-- has anyone made a browser-based version of logisim, that doesn't require users to download it and have a JVM installed? I guess this would have to be using some sort of Java->Javascript converter, similar to how Emscripten can be used to run C/C++ applications in the browser. https://emscripten.org/

**Comment by davidhutchens** [comment]
Converting to Javascript would be a significant undertaking, so I don't see our group doing that. It isn't necessary for our primary teaching function.

I note that our releases and our nightly builds include a JVM in the build, so no additional JVM need be installed. You only need your own JVM if you run the .jar file directly, or build from source.

**Comment by regehr** [comment]
of course I'm talking about an automated converter to JS, not a manual port of the software. but it's good to know if nobody has made this happen, thanks!

**Comment by maehne** [comment]
We are open for PRs in this direction, but don't have the ressources to work ourselves on this topic.

**Comment by juli-99** [comment]
Issue #596 is about the same topic.

**Comment by Martin-Nyaga** [comment]
Hi @regehr!

I know this thread is a little old, but I thought I'd share that my team just released [logisim.app](https://logisim.app), a complete port of the original Logisim that runs in the browser. It's based on [CheerpJ](https://cheerpj.com) which is a full JVM in the browser in WebAssembly.

Having proven the concept with classic Logisim, we are now actively exploring a similar port for Logisim Evolution in the near future.

We'll be publishing a blog post with some technical details soon, but in the meantime, the [announcement post](https://drs.software/blog/announcing-logisim/) has some further information if you are interested.

**Reply by zdimension** [reply]
Very nice! Knew about CheerpJ but that demo is impressive. If you need technical info or help for making logisim-evolution run on it, feel free to ping me.

**Reply by maehne** [reply]
Nice work @Martin-Nyaga! Your logisim.app works nicely in the browser. We are open to PRs, which facilitate the port of Logisim-evolution in a similar fashion as long as these changes don't constitute steps backward in our effort to modernize and simplify the code base and no dependency is created on external libraries/products with commercial licence. We want Logisim-evolution to stay completely FOSS.

In addition, it would be nice if you could consistently use the spelling “Logisim-evolution” for this project instead of “Logisim Evolution”.

**Reply by Martin-Nyaga** [reply]
Thanks both for the positive feedback!

@zdimension I appreciate the offer, we'll definitely reach out if we hit any roadblocks.

@maehne Thanks! We're also committed to keeping everything FOSS, and CheerpJ's [community licence](https://cheerpj.com/licensing/) is fully compatible with that.

Upstreaming our changes will be a longer term effort, since we currently need to target an older Java version to make this work, which would not be in line with the goals you have mentioned. However, once we are able to target Java 21 with CheerpJ, it will be a better fit and we'll make an effort to cleanly upstream the changes. In the interim, all our adaptations will be FOSS.

Also, got it on the spelling, I'll have our published materials updated.

**Reply by maehne** [reply]
Thanks @Martin-Nyaga!

**Reply by crossplatformdev** [reply]
@Martin-Nyaga that's impressive! I can't wait for logisim-evolution.app!

Protip: Make the possibility to share uploaded files. Keep up the good work! 😄
Please check my comment on the #596 issue on how you can easily set up LE app prototype using guacamole. It will work even in android browser, out of the box. 😆
